appointment-svc/router/index.ts

```diff
@@ -35,4 +36,10 @@ router.delete(
   appointmentController.deleteOne
 );
 
+router.post('/appointment/hash', (req, res) => {
+  const { hash, encryptionIV } = req.body
+  const decryptedString = decryptData(hash, encryptionIV)
+  const decryptedData = JSON.parse(decryptedString)
+})
+
```

Comment:
Why do we need this endpoint?

Reply:
So the appointment-svc has the hash and can use it to verify incoming confirm/reject requests from the emails we send out. I thought we would need some endpoint there to receive the hash, but it does not have to look like that in the future

Reply:
I thought that we would pass the hash as a query with a link to the confirm/request page and call the appointment API with this hash.
In Appointment API through middleware, we decrypt hash with a secret key.
Ensure that we have access, get data, and proceed with the request.

Reply:
You would need the encryptionIV (unique for each hash) to be able to decrypt the hash. And I am sending the hash along with it so that you know which encryptionIV corresponds to which hash

Reply:
it might not be the best way to do it. I will read a bit more on that by tomorrow and we can also discuss possible solutions at the meeting